.github/workflows/pr_test_cpu.yml

```diff
@@ -0,0 +1,49 @@
+name: Run tests on CPU
+
+on:
+  pull_request:
+    branches: [master]
+    types: [open, reopened, synchronize, review_requested, ready_for_review]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  tests-cpu:
+    name: ${{ matrix.os }} - python-${{ matrix.python-version }}, torch-${{ matrix.pytorch-version }}, ${{ matrix.pytorch-dtype }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos]
+        python-version: ['3.7', '3.8']
+        pytorch-version: ['1.9.1', '1.12.1']
+        pytorch-dtype: ['float16', 'float32', 'float64']
+
+    env:
+      continueIfFails: ${{ contains('float16', matrix.pytorch-dtype) }} # Remove this when float16 stop raising errors
+
+    steps:
+      - name: Checkout kornia
+        uses: actions/checkout@v3
+
+      - name: Setting environment on ${{ matrix.os }} with python ${{ matrix.python-version }} and pytorch  ${{ matrix.pytorch-version }}
+        uses: ./.github/actions/env
+        with:
+          python-version: ${{ matrix.python-version }}
+          pytorch-version: ${{ matrix.pytorch-version }}
+
+      - name: Run CPU tests ${{ matrix.pytorch-dtype }}
+        continue-on-error: ${{ env.continueIfFails }}
+        shell: bash -l {0}
+        run: pytest -v --device cpu --dtype ${{ matrix.pytorch-dtype }} --cov=kornia --cov-report xml ./test
```

Comment:
Maybe add `  -x, --exitfirst       Exit instantly on first error or failed test
` to pytest?

Reply:
Let’s add if we see that the building is too long